Sources/BraintreePaymentFlow/BTPaymentFlowDriver.m

```diff
@@ -123,21 +78,53 @@ - (void)onPaymentWithURL:(NSURL *)url error:(NSError *)error {
     }
     [self.apiClient sendAnalyticsEvent:[NSString stringWithFormat:@"ios.%@.webswitch.initiate.succeeded", [self.paymentFlowRequestDelegate paymentFlowName]]];
 
-    [self performSwitchRequest:url];
-}
+    self.authenticationSession = [[ASWebAuthenticationSession alloc] initWithURL:url callbackURLScheme:BTCallbackURLScheme completionHandler:^(NSURL * _Nullable callbackURL, NSError * _Nullable error) {
+        // Required to avoid memory leak for BTPaymentFlowDriver
+        self.authenticationSession = nil;
+
+        if (error) {
+            if (error.domain == ASWebAuthenticationSessionErrorDomain && error.code == ASWebAuthenticationSessionErrorCodeCanceledLogin) {
+                [self.apiClient sendAnalyticsEvent:[NSString stringWithFormat:@"ios.%@.webswitch.canceled", [self.paymentFlowRequestDelegate paymentFlowName]]];
+            }
+            NSError *err = [NSError errorWithDomain:BTPaymentFlowDriverErrorDomain
+                                               code:BTPaymentFlowDriverErrorTypeCanceled
+                                           userInfo:@{NSLocalizedDescriptionKey: @"Payment flow was canceled by the user."}];
+            self.paymentFlowCompletionBlock(nil, err);
+            paymentFlowDriver = nil;
+            return;
+        }
+
+        [self.apiClient sendAnalyticsEvent:[NSString stringWithFormat:@"ios.%@.webswitch.succeeded", [self.paymentFlowRequestDelegate paymentFlowName]]];
+
+        [self.paymentFlowRequestDelegate handleOpenURL:callbackURL];
+    }];
+
+    if (@available(iOS 13, *)) {
+        self.authenticationSession.presentationContextProvider = self;
+    }
 
-- (void)onPaymentCancel {
-    [self.apiClient sendAnalyticsEvent:[NSString stringWithFormat:@"ios.%@.webswitch.canceled", [self.paymentFlowRequestDelegate paymentFlowName]]];
-    NSError *error = [NSError errorWithDomain:BTPaymentFlowDriverErrorDomain
-                                         code:BTPaymentFlowDriverErrorTypeCanceled
-                                     userInfo:@{NSLocalizedDescriptionKey: @"Payment flow was canceled by the user."}];
-    self.paymentFlowCompletionBlock(nil, error);
-    paymentFlowDriver = nil;
+    [self.authenticationSession start]; // TODO what if the start fails
 }
 
 - (void)onPaymentComplete:(BTPaymentFlowResult *)result error:(NSError *)error {
     self.paymentFlowCompletionBlock(result, error);
     paymentFlowDriver = nil;
 }
 
+#pragma mark - ASWebAuthenticationPresentationContextProviding protocol
+
+- (ASPresentationAnchor)presentationAnchorForWebAuthenticationSession:(ASWebAuthenticationSession *)session API_AVAILABLE(ios(13)) {
```

Comment:
Probably safer if we have the merchant pass us their active scene versus us fetching it. Check what the PayPalDriver is doing.